spec/views/components/autocomplete_spec.rb

```diff
@@ -0,0 +1,90 @@
+require 'rails_helper'
+
+RSpec.describe "Autocomplete component" do
+  it "should not render when no data is given" do
+    assert_empty render_component({})
+  end
+
+  it "should not render when only a name is provided" do
+    assert_empty render_component(name: "foo")
+  end
+
+  it "should not render when only a label is provided" do
+    assert_empty render_component(label: { text: "foo" })
+  end
+
+  it "should render text input with randomised id when name & label provided" do
+    render_component(name: "foo", label: { text: "bar" })
+
+    assert_select ".app-c-autocomplete", count: 1
+    randomised_for = nil
+    assert_select ".app-c-autocomplete label", count: 1, text: "bar" do |label|
+      randomised_for = label[0]['for']
```

Comment:
What is this?

Reply:
The label has a randomised `for` attribute (e.g. `<label for="autocomplete-abc123">`) if no `id` is provided.

It's important that the `for` attribute of the label matches with the `id` attribute of the input so that it is accessible. I could probably write it a bit nicer though 🤔 will have a think...

Reply:
Have rewritten the test name and the variable name, hopefully clearer.